lib/MockKubeResourceMeta.js

```diff
@@ -0,0 +1,69 @@
+module.exports = class MockKubeResourceMeta {
+  constructor(apiVersion, kind, kubeData) {
+    this._apiVersion = ((apiVersion === undefined || apiVersion === '') ? 'deploy.razee.io/v1alpha2' : apiVersion);
+    this._kind = ((kind === undefined || kind === '') ? 'MustacheTemplate' : kind);
+    this.kubeData = kubeData;
+  }
+
+  async request(reqOpt) {
+    const ref = {
+      apiVersion: reqOpt.uri.apiVersion,
+      kind: reqOpt.uri.kind,
+      name: reqOpt.uri.name,
+      namespace: reqOpt.uri.namespace,
+      labelSelector: reqOpt?.qs?.labelSelector
+    };
+
+    const res = await this.kubeGetResource(ref);
+    return res;
+  }
+
+  async get(name, namespace) {
+    const ref = this.uri({ name, namespace });
+    return await this.kubeGetResource(ref);
+  }
+
+  uri(options) {
+    return { ...options, apiVersion: this._apiVersion, kind: this._kind, };
+  }
+
+  async kubeGetResource(ref) {
+    const {
+      name,
+      labelSelector,
+      namespace,
+      kind,
+      apiVersion
+    } = ref;
+    if (!this.kubeData[kind]) {
+      return;
+    }
+
+    let fn = labelSelector ? 'filter' : 'find';
+    let lookup = this.kubeData[kind][fn](obj => {
+      let match = true;
+      match = (obj.apiVersion === apiVersion && match) ? true : false;
+      match = (obj.kind === kind && match) ? true : false;
+      match = ((obj.metadata.name === name || labelSelector !== undefined) && match) ? true : false;
+      // match = (obj.metadata.namespace === namespace && match) ? true : false;
+      // ^this is the original check for namespace,
+      //  but it doesn't account for no namespace in the obj.metadata when
+      //  looking for default namespace
+      match = (
+        (
+          obj.metadata.namespace === namespace ||
+         (obj.metadata.namespace === undefined && namespace === 'default')
+        ) && match
+      ) ? true : false;
```

Comment:
its not normally up to the KRM to handle wether a namespace is defined, so i find this a bit confusing from a code perspective. I assume you added it because you have resources that use this mock that dont define a namespace.

Id argue since the code doesnt allow you to use KRM without a namespace defined, for testing you shouldnt be allowing it either. also we dont default the namespace to the `default` namespace anyways, so it could be misleading to allow no namespace to translate to `default` namespace.

Reply:
ah, interesting.

yeah, we have configmaps w/o the namespace defined, because when the remote resource controller applies them w/o namespace, k8s puts them in default.

when i first got this working, i lost about a day trying to figure out why only 1 of the 3 CMs my MTP was supposed to be importing were getting rendered, before i realized it was because only 1/3 had a namespace defined.

i can pull this out, but if it isn't going to have a fallback to the default namespace, then i think at least a warning that an undefined namespace causes unexpected behavior would be good. i could see a lot more time lost on debugging if there isn't _something_ to explain to the user why their CMs aren't being rendered

Reply:
FWIW: if no namespace is defined, RR Controller will use the namespace of the remoteResource that created the child resource https://github.com/razee-io/razeedeploy-core/blob/master/lib/CompositeController.js#L149-L152